### PR TITLE
[alertmanager-snmp-notifier] add SNMP notifer chart

### PR DESCRIPTION
**Pull request by maxwo**
#### What this PR does / why we need it

This PR creates a SNMP notifier chart, for creating an alertmanager webhook that sends alerts as SNMP traps.

#### Special notes for your reviewer

I created a previous PR, but closed and created a new one, squashed, and up-to-date with a newest version.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/alertmanager-snmp-notifier/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Prometheus SNMP Notifier
+
+An Alertmanager webhook that relays alerts as SNMP traps.
+
+This chart creates a [SNMP Notifier](https://github.com/maxwo/snmp_notifier) deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
+
+## Prerequisites
+
+- Kubernetes 1.8+ with Beta APIs enabled
```
Comment (monotek) [inline charts/alertmanager-snmp-notifier/README.md:9]: This seems kinda old. 
Was this testesd onm this old version?
I would rather start with some k8s version which is not already eol...

File: charts/alertmanager-snmp-notifier/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+_See [helm repo](https://helm.sh/docs/helm/helm_repo/) for command documentation._
+
+## Install Chart
+
+```console
+# Helm 3
+$ helm install [RELEASE_NAME] prometheus-community/snmp-notifier
+
+# Helm 2
```
Comment (monotek) [inline charts/alertmanager-snmp-notifier/README.md:26]: helm 2 is deprecated. please remove all occurrences.

File: charts/alertmanager-snmp-notifier/values.yaml [hunk 3]
```diff
@@ -0,0 +1,185 @@
+restartPolicy: Always
+
+kind: Deployment
+
+image:
+  repository: maxwo/snmp-notifier
+  tag: v1.4.0
```
Comment (monotek) [inline charts/alertmanager-snmp-notifier/values.yaml:7]: i would rather use appversion var from Chart.yaml.

Example:

https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-conntrack-stats-exporter/values.yaml#L6-L7
https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-conntrack-stats-exporter/templates/daemonset.yaml#L35

File: charts/alertmanager-snmp-notifier/values.yaml [hunk 4]
```diff
@@ -0,0 +1,185 @@
+restartPolicy: Always
```
Comment (monotek) [inline charts/alertmanager-snmp-notifier/values.yaml:1]: please sort all vars alphabetically.

File: charts/alertmanager-snmp-notifier/values.yaml [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+  name:
+
+resources:
+  {}
+  # limits:
+  #   memory: 300Mi
+  # requests:
+  #   memory: 50Mi
+
+livenessProbe:
```
Comment (monotek) [inline charts/alertmanager-snmp-notifier/values.yaml:128]: liveness and readinessprobe should not be the same.
see: https://srcco.de/posts/kubernetes-liveness-probes-are-dangerous.html

File: charts/alertmanager-snmp-notifier/values.yaml [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+service:
+  annotations: {}
+  type: ClusterIP
+  port: 9464
+
+## An Ingress resource can provide name-based virtual hosting and TLS
+## termination among other things for CouchDB deployments which are accessed
+## from outside the Kubernetes cluster.
+## ref: https://kubernetes.io/docs/concepts/services-networking/ingress/
+ingress:
```
Comment (monotek) [inline charts/alertmanager-snmp-notifier/values.yaml:146]: this seems outdated. 
have you created the chart by using a recent helm version?

if not please recreate the chart via:
```
helm create alertmanager-snmp-notifier
```

File: charts/alertmanager-snmp-notifier/values.yaml [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+
+podAnnotations: {}
+
+replicas: 1
+
+# Enable this if you're using https://github.com/coreos/prometheus-operator
+serviceMonitor:
+  # When set true then use a ServiceMonitor to configure scraping
+  enabled: false
+  # Set the namespace the ServiceMonitor should be deployed
```
Comment (monotek) [inline charts/alertmanager-snmp-notifier/values.yaml:170]: values dont have to be commented out if enbaled is false anyway.

**Review by monotek — CHANGES_REQUESTED** [review]
please also add yourself to the codeowers file: https://github.com/prometheus-community/helm-charts/blob/main/.github/CODEOWNERS

File: charts/alertmanager-snmp-notifier/README.md (hunk 1 above)
Comment (maxwo) [reply charts/alertmanager-snmp-notifier/README.md]: Indeed. I reused another chart as a base in that same repository, thinking I would comply with some rules of this helm repository.
I changed it with the version I could properly test, but it should be compatible with much older versions.

File: charts/alertmanager-snmp-notifier/README.md (hunk 2 above)
Comment (maxwo) [reply charts/alertmanager-snmp-notifier/README.md]: Done.

File: charts/alertmanager-snmp-notifier/values.yaml (hunk 3 above)
Comment (maxwo) [reply charts/alertmanager-snmp-notifier/values.yaml]: Done, thanks for the info!

File: charts/alertmanager-snmp-notifier/values.yaml (hunk 4 above)
Comment (maxwo) [reply charts/alertmanager-snmp-notifier/values.yaml]: Fixed, although I kept the `helm create` order if that is ok for you.

File: charts/alertmanager-snmp-notifier/values.yaml (hunk 5 above)
Comment (maxwo) [reply charts/alertmanager-snmp-notifier/values.yaml]: I am aware of this, but the app won't launch, if the SNMP server is unreachable or if there is a bad option, or bad template provided.
The HTTP server is the latest thing that is launched if everything is OK. That /health endpoint is therefore "the app is well configured and ready to receive alerts". I don't see what else could I check to differentiate the liveness from readiness.

File: charts/alertmanager-snmp-notifier/values.yaml (hunk 6 above)
Comment (maxwo) [reply charts/alertmanager-snmp-notifier/values.yaml:97]: Indeed, I recreated the chart using this command. Thanks.

File: charts/alertmanager-snmp-notifier/values.yaml (hunk 7 above)
Comment (maxwo) [reply charts/alertmanager-snmp-notifier/values.yaml:142]: Indeed, this is fixed.

**Comment by maxwo** [comment]
I just taken into account your logs about the PR. Thanks for all the infos, I could make some things much cleaner and up-to-date.
I was wondering if `alertmanager-snmp-notifier` is a good name though, I hesitated with `snmp-notifier` only, but wanted to make it clear it is not a proper exporter.

**Review by monotek — CHANGES_REQUESTED** [review]
please fix ci

**Review by monotek — APPROVED** [review]

**Pull request merged**